### PR TITLE
Display created chapters in details

### PR DESCRIPTION
**Pull request by SkyDiverStar**
# Description of changes

Added code in [course]/index.tsx for proper display of the chapters, that should be visibile.

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: pages/course/[courseId]/index.tsx [hunk 1]
```diff
@@ -27,22 +35,24 @@ export default function Details() {
   const course = coursesById[0];
 
   //TODO: change later, when implementing the services into this side
```
Comment (mardom1) [inline pages/course/[courseId]/index.tsx:37]: This ran be removed now, right?

File: pages/course/[courseId]/index.tsx [hunk 2]
```diff
@@ -27,22 +35,24 @@ export default function Details() {
...
   const usage = 22;
+  let viewablechapters = [];
+
+  for (let index = 0; index < chapters.length; index++) {
+    const element = chapters.at(index);
+    const start = new Date(element?.startDate);
+    const end = new Date(element?.endDate);
+    if (
+      start.toISOString() <= currentDate.toISOString() &&
+      end.toISOString() >= currentDate.toISOString()
```
Comment (mardom1) [inline pages/course/[courseId]/index.tsx:51]: Comparing them directly (`start <= currentdate`) should be better than string comparison.

File: pages/course/[courseId]/index.tsx [hunk 3]
```diff
@@ -27,22 +35,24 @@ export default function Details() {
...
-    "Chapter 10",
-  ];
+  const chapters = course.chapters;
   const knowledge = 88;
   const understanding = 50;
   const analyses = 40;
   const usage = 22;
+  let viewablechapters = [];
+
+  for (let index = 0; index < chapters.length; index++) {
```
Comment (mardom1) [inline pages/course/[courseId]/index.tsx:45]: You can also use `for (let chapter in chapters) {...}` (if you didn't know).
Or even `const viewablechapters = chapters.filter(chapter => ...)`.

Just a comment though. This is also fine for me.

**Review by mardom1 — CHANGES_REQUESTED** [review]
Apart from the date string comparison, it looks good

File: pages/course/[courseId]/index.tsx (hunk 2 above)
Comment (SkyDiverStar) [reply pages/course/[courseId]/index.tsx:51]: I based it on following [article](https://programmingwithswift.com/how-to-compare-dates-with-typescript/). There they tested it with string and with normal date comparison, but the normal comparison did not work correctly. But if this article is outdate I can change it.

File: pages/course/[courseId]/index.tsx (hunk 3 above)
Comment (SkyDiverStar) [reply pages/course/[courseId]/index.tsx:45]: The first option: I forgot about that possibility, thanks for the reminder.
The second option: I tried that at first, but I did not get it to work, so I used the old school method.

File: pages/course/[courseId]/index.tsx (hunk 1 above)
Comment (SkyDiverStar) [reply pages/course/[courseId]/index.tsx:37]: You mean the TODO? Not really, because it has still hardcoded numbers for the skill level, because we still do not have the skill level application.

File: pages/course/[courseId]/index.tsx (hunk 2 above)
Comment (mardom1) [reply pages/course/[courseId]/index.tsx:51]: I guess I'd prefer `.getTime()`, but lets get this merged.l as it's personal preference.

**Review by mardom1 — APPROVED** [review]
